src/geometry/ArborX_DetailsAlgorithms.hpp

```diff
@@ -573,6 +574,165 @@ struct intersects<PointTag, TriangleTag, Point, Triangle>
   }
 };
 
+template <typename Box, typename Triangle>
+struct intersects<BoxTag, TriangleTag, Box, Triangle>
+{
+  KOKKOS_FUNCTION static constexpr bool apply(Box const &box,
+                                              Triangle const &triangle)
+  {
+    // Based on the Separating Axis Theorem
+
+    // Test bounding boxes, i.e., normals of the box
+    Box triangle_bounding_box;
+    ArborX::Details::expand(triangle_bounding_box, triangle);
+    if (!ArborX::Details::intersects(triangle_bounding_box, box))
+      return false;
+
+    // shift to origin
+    constexpr int DIM = GeometryTraits::dimension_v<Triangle>;
+    static_assert(DIM == 3,
+                  "Box-Triangle intersection only implemented in 3d!");
+    using Float = typename GeometryTraits::coordinate_type_t<Triangle>;
+    auto min_corner = box.minCorner();
+    auto max_corner = box.maxCorner();
+    auto triangle_a = triangle.a;
+    auto triangle_b = triangle.b;
+    auto triangle_c = triangle.c;
+    ExperimentalHyperGeometry::Point<DIM, Float> edge_ab;
+    ExperimentalHyperGeometry::Point<DIM, Float> edge_ac;
+    ExperimentalHyperGeometry::Point<DIM, Float> extents;
+    for (int i = 0; i < DIM; ++i)
+    {
+      Float const shift = -(max_corner[i] + min_corner[i]) / 2;
+      min_corner[i] += shift;
+      max_corner[i] += shift;
+      triangle_a[i] += shift;
+      triangle_b[i] += shift;
+      triangle_c[i] += shift;
+      edge_ab[i] = triangle_b[i] - triangle_a[i];
+      edge_ac[i] = triangle_c[i] - triangle_a[i];
+      extents[i] = (max_corner[i] - min_corner[i]) / 2;
+    }
+
+    // Test normal of the triangle
+    ExperimentalHyperGeometry::Point<DIM, Float> normal{
+        {edge_ab[1] * edge_ac[2] - edge_ab[2] * edge_ac[1],
+         edge_ab[2] * edge_ac[0] - edge_ab[0] * edge_ac[2],
+         edge_ab[0] * edge_ac[1] - edge_ab[1] * edge_ac[0]}};
+    Float radius = extents[0] * Kokkos::abs(normal[0]) +
+                   extents[1] * Kokkos::abs(normal[1]) +
+                   extents[2] * Kokkos::abs(normal[2]);
+    Float triangle_a_projected = triangle_a[0] * normal[0] +
+                                 triangle_a[1] * normal[1] +
+                                 triangle_a[2] * normal[2];
+    if (Kokkos::abs(triangle_a_projected) > radius)
+      return false;
+
+    // Test crossproducts
+    ExperimentalHyperGeometry::Point<DIM, Float> edge_bc{
+        triangle_c[0] - triangle_b[0], triangle_c[1] - triangle_b[1],
+        triangle_c[2] - triangle_b[2]};
+
+    // e_x x edge_ab = (0, -edge_ab[2],  edge_ab[1])
+    {
+      Float radius = extents[1] * Kokkos::abs(edge_ab[2]) +
+                     extents[2] * Kokkos::abs(edge_ab[1]);
+      Float xab_0 = -triangle_a[1] * edge_ab[2] + triangle_a[2] * edge_ab[1];
+      Float xab_1 = -triangle_c[1] * edge_ab[2] + triangle_c[2] * edge_ab[1];
+      if (Kokkos::fmin(xab_0, xab_1) > radius ||
+          Kokkos::fmax(xab_0, xab_1) < -radius)
+        return false;
+    }
```

Comment:
This and similar codes are pretty hard to read. Maybe a helper function that would encapsulate it? I also can't understand what's going on from the comment. It would help to provide more context.